### PR TITLE
polymc: Use portable configuration

### PR DESCRIPTION
**Pull request by sitiom**
Reverts #674.

By principle, Scoop should use portable versions if possible. Otherwise, they should be in the [Nonportable](https://github.com/ScoopInstaller/Nonportable) bucket. This has already been applied to the other official buckets (https://github.com/ScoopInstaller/Extras/pull/7573), the games bucket should follow too.

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sitiom** [comment]
A migration script would keep current users happy (https://github.com/Calinou/scoop-games/pull/674#issuecomment-1207192740).

**Comment by sitiom** [comment]
@Calinou All good. Run the migration script to migrate your config to Scoop:
image

Ping @dada513 @DioEgizio @Pato05 @txtsd

**Comment by DioEgizio** [comment]
polymc-qt5 should be adapted to this too

File: bucket/polymc.json [hunk 1]
```diff
@@ -3,19 +3,40 @@
...
+        "accounts",
+        "assets",
+        "instances",
+        "libraries",
+        "meta",
+        "themes",
+        "translations",
+        "accounts.json",
+        "polymc.cfg",
+        "notifications.json"
```
Comment (DioEgizio) [inline bucket/polymc.json:34]: notifications don't exist in polymc anymore for a long time

File: scripts/polymc/migrate.ps1 [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "accounts",
+    "assets",
+    "instances",
+    "libraries",
+    "meta",
+    "themes",
+    "translations",
+    "accounts.json",
+    "polymc.cfg",
+    "notifications.json"
```
Comment (DioEgizio) [inline scripts/polymc/migrate.ps1:11]: .

**Comment by sitiom** [comment]
> polymc-qt5 should be adapted to this too

@DioEgizio I'm assuming the persist files are the same in the legacy version, right? They seem to be missing in the manifest.

File: scripts/polymc/migrate.ps1 [hunk 3]
```diff
@@ -0,0 +1,27 @@
+$files = @(
+    "accounts",
+    "assets",
+    "instances",
+    "libraries",
+    "meta",
+    "themes",
+    "translations",
+    "accounts.json",
+    "polymc.cfg",
```
Comment (DioEgizio) [inline scripts/polymc/migrate.ps1:10]:
```suggestion
    "polymc.cfg"
```

I assume

**Comment by DioEgizio** [comment]
> > polymc-qt5 should be adapted to this too
> 
> @DioEgizio I'm assuming the persist files are the same in the legacy version, right? They seem to be missing in that manifest.

Right now polymc-qt5 uses the not portable build, but the files stay in %APPDATA%/PolyMC too, so the migration script should be the same

File: scripts/polymc/migrate.ps1 (hunk 3 above)
Comment (sitiom) [reply scripts/polymc/migrate.ps1:10]: Oops

**Comment by sitiom** [comment]
> Right now polymc-qt5 uses the not portable build, but the files stay in %APPDATA%/PolyMC too, so the migration script should be the same

Are you sure? It has been using the [portable version](https://github.com/Calinou/scoop-games/blob/3fe019f9b03041fabee8e7b004f6a48ff6578590/bucket/polymc-qt5.json#L6) from the start, so I don't think a migration script is needed for that. I'm just concerned about the lack of persisted files in that manifest.

**Comment by sitiom** [comment]
After some thought, I would prefer using the non-portable config version because if PolyMC is installed globally, you would need admin permissions to write to files every time.
image

**Comment by Pato05** [comment]
@sitiom maybe keeping it non-portable for global installs?

Shouldn't be hard since I saw that PolyMC enables portable mode when there's a "portable.txt" file in its root directory.

It would be better keeping all data in a predictable directory to make uninstallation easier, instead of everything scrambled in the file system.

**Comment by DioEgizio** [comment]
Yes, the actual difference between portable and not portable configurations is managed by the portable.txt file; if it's there is portable, if it's not there it's not portable

**Comment by sitiom** [comment]
> @sitiom maybe keeping it non-portable for global installs?

> It would be better keeping all data in a predictable directory to make uninstallation easier, instead of everything scrambled in the file system.

If you make the global installation non-portable and the local portable, that would be scrambled, wouldn't it?

**Comment by Pato05** [comment]
> > @sitiom maybe keeping it non-portable for global installs?
> 
> > It would be better keeping all data in a predictable directory to make uninstallation easier, instead of everything scrambled in the file system.
> 
> If you make the global installation non-portable and the local portable, that would be scrambled, wouldn't it?

yes but most users would install locally anyways, and you could add a note for that, to warn the user

**Comment by sitiom** [comment]
> this would resolve the problem for local installs

It's not worth the inconsistencies that it will bring between the local and global installations. The current setup works well, config files are located at `%appdata%\PolyMC` regardless of whether it is local or global.

**Comment by Calinou** [comment]
I think it's better to keep polymc non-portable in this case.

**Pull request closed without merging**